src/optimizer.rs

```diff
@@ -102,8 +102,7 @@ impl Optimization for SimplePatternGroup {
             }
         }
 
-        // let is_regex = filters.iter().find(|f| f.is_regex()).is_some();
```

Comment:
Any idea why it was disabled?
I also believe we shouldn't fuse regex with non regex rules.
So a possible solution is returning a pair or a vector `[fused_non_regex_rules, fused_regex_rules]`

Reply:
https://github.com/brave/adblock-rust/commit/5c28aefc#diff-40a35a91a642ade2cf6b8ce86ddb3c8b82625c0592d0e9fa24d2b41a072ab3adR92

No idea :stuck_out_tongue: